src/main/java/mate/jdbc/dao/impl/CarDaoImpl.java

```diff
@@ -0,0 +1,222 @@
+package mate.jdbc.dao.impl;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.jdbc.dao.CarDao;
+import mate.jdbc.exception.DataProcessingException;
+import mate.jdbc.lib.Dao;
+import mate.jdbc.model.Car;
+import mate.jdbc.model.Driver;
+import mate.jdbc.model.Manufacturer;
+import mate.jdbc.util.ConnectionUtil;
+
+@Dao
+public class CarDaoImpl implements CarDao {
+
+    @Override
+    public Car create(Car car) {
+        String query = "INSERT INTO cars (`model`, `manufacturers_id`) VALUES (?, ?);";
+        try (Connection connection = ConnectionUtil.getConnection();
+                    PreparedStatement statement = connection.prepareStatement(query,
+                            Statement.RETURN_GENERATED_KEYS)) {
+            statement.setString(1, car.getModel());
+            statement.setLong(2, car.getManufacturer().getId());
+            statement.executeUpdate();
+            ResultSet resultSet = statement.getGeneratedKeys();
+            if (resultSet.next()) {
+                car.setId(resultSet.getObject(1, Long.class));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't create car " + car, e);
+        }
+        return insertDriversToCar(car);
+    }
+
+    @Override
+    public Optional<Car> get(Long id) {
+        String query = "SELECT model, car_id, name, country, manufacturers_id "
+                + "FROM cars c "
+                + "INNER JOIN cars_drivers cd "
+                + "ON cd.car_id = c.id "
+                + "INNER JOIN manufacturers m "
+                + "ON c.manufacturers_id = m.id "
+                + "WHERE car_id = ? AND c.is_deleted = FALSE;";
+        Car car = null;
+        try (Connection connection = ConnectionUtil.getConnection();
+                            PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                car = getCarWithoutListOfDrivers(resultSet);
+                car.setDrivers(getDriverList(car.getId()));
```

Comment:
i like ur solution but let's make few changes
 in  car.setDrivers(getDriverList(car.getId())); u open new connection 
 it will be better use it after try catch 
 better tou use it before return statement